### DISCUSSION TITLE
Font Size Inheritance Issue with PrimeNG 19 Components

### DISCUSSION THREAD

**Discussion by yanivbrik** [primefaces/community#3273] [PrimeNG]
I noticed an inconsistency in font size inheritance when using PrimeNG components. 
According to the theming documentation:

"There is no design for fonts as UI components inherit their font settings from the application."

I have set up my base font sizing as follows:

html {
  font-size: 62.5%;
}

body {
  font-family: /* family name */;
  font-size: 1.4rem;
}

**Issue:** I've observed that PrimeNG components use rem units instead of em for font sizes. For example, .p-inputtext has font-size: 1rem instead of font-size: 1em. This affects the expected inheritance behavior.

**Current Workaround:** I'm currently fixing this by overriding each component:

body .p-inputtext {
  font-size: 1em;
}

**Questions:**

Is this the intended behavior?
Should the Aura theme design tokens use em instead of rem for better component inheritance?
Is there a recommended approach to handle font scaling without having to override each component?

**Comment by qwadrox** [comment]
@yanivbrik 

They deliberately using `rem` I think. if you read this blog post you will understand the differences between them: https://refine.dev/blog/rem-vs-em/#compound-scaling-with-em

`em` units can compound when nested, as each nested element's font size is calculated based on its immediate parent's font size. This can make them harder to predict:

```css
.parent {
    font-size: 16px;
}
.child {
    font-size: 2em;    /* 32px */
}
.grandchild {
    font-size: 2em;    /* 64px! (2 × 32px) */
}
```

<img width="372" alt="image" src="https://github.com/user-attachments/assets/1eef4b4f-2212-45d1-b195-48133ce20afe" />

So `rem` is better for font sizes and general spacing to maintain consistency. You set the ratio in the root HTML tag and it will be inheretied from there. 

https://primeng.org/theming#scale

Basically you can scale up and down everything by modifying one variable.

**Comment by yanivbrik** [comment]
I appreciate the explanation, but my use case is different. Let me clarify:

I'm using this specific setup:

html {
  font-size: 62.5%;  /* Makes 1rem = 10px for easier calculations */
}

body {
  font-size: 1.4rem; /* Results in 14px default size */
}

This approach provides two key benefits:

It makes rem calculations intuitive (e.g., 5rem = 50px)
It maintains proper scaling when users change their browser's default font size
The compounding issue with em units you mentioned isn't relevant here, as I'm specifically using this setup for predictable scaling across the entire application.

**Comment by Buk1m** [comment]
This is in my opinion how libraries should define components to allow for best of both worlds. It is problematic now to add primeng to existing project that is based on default html font-size: 16px;

https://svelte.dev/playground/f182f2ae77614b7cbb8aaa7dbc5067ab?version=5.37.1
